hypershift-operator/metrics.go

```diff
@@ -156,11 +210,24 @@ func (m *hypershiftMetrics) collect(ctx context.Context) error {
 	if err := m.observeNodePools(ctx, &nodePools); err != nil {
 		return err
 	}
+
 	return nil
 }
 
 func setupMetrics(mgr manager.Manager) error {
-	metrics := newMetrics(mgr.GetClient(), mgr.GetLogger().WithName("metrics"))
+	var hypershiftImage ImageInfo
+
+	// We need to create a new client because the manager one still does not have the cache started
+	tmpClient, err := crclient.New(mgr.GetConfig(), crclient.Options{Scheme: hyperapi.Scheme})
+	if err != nil {
+		return fmt.Errorf("error creating a temporary client: %v\n", err)
+	}
+	// Grabbing the Image and ImageID from Operator
+	if hypershiftImage.image, hypershiftImage.imageId, err = getOperatorImage(tmpClient); err != nil {
```

Comment:
the way this is now it's going to fail and return here every time the pod doesn't exist, e.g. in dev env. You might want to log and let the workflow move on when the error is pod is not found